### PR TITLE
bpo-40928: notify users of osx malloc warnings

### PR DESCRIPTION
**Pull request by jdevries3133**
This is a bandaid to help people save time debugging this non-bug. I don't think this solution is very strong; I'm just hoping to move the discussion forward on the bpo, and hoping for a hint on how to suppress these warnings all together or come up with some other optimal solution.

I applied the function in test_decimal because that test consistently generates these malloc warnings on my system.

https://bugs.python.org/issue40928

### CODE REVIEW

**Comment by jdevries3133** [comment]
This is one of two potential fixes I've provided for bpo-40928. The other fix, #26786, gets more to the root of the problem. This is more of a simple band-aid.

File: Lib/test/support/__init__.py [hunk 1]
```diff
@@ -448,6 +448,22 @@ def requires_lzma(reason='requires lzma'):
 TEST_DATA_DIR = os.path.join(TEST_HOME_DIR, "data")
 
 
+def darwin_malloc_err_warning():
+    """Assure user that loud errors generated by OS X libc's malloc are
+    expected."""
+    if sys.platform != 'darwin':
+        return
+    msg = ' NOTICE '
+    detail = ('test_decimal may generate "malloc can\'t allocate region"\n'
```
Comment (ambv) [inline Lib/test/support/__init__.py]: Extract `test_decimal` to a function argument called "test_name".

File: Lib/test/support/__init__.py [hunk 2]
```diff
@@ -448,6 +448,22 @@ def requires_lzma(reason='requires lzma'):
 TEST_DATA_DIR = os.path.join(TEST_HOME_DIR, "data")
 
 
+def darwin_malloc_err_warning():
+    """Assure user that loud errors generated by OS X libc's malloc are
```
Comment (ambv) [inline Lib/test/support/__init__.py]: There is no "OS X", change all such usage to "macOS" instead.

**Review by ambv — CHANGES_REQUESTED** [review]
This is a safe change to introduce regardless of what we do with the root cause.

**Comment by jdevries3133** [comment]
@ambv Ok, your suggested changes were incorporated in 4859c7ef7526d4a0d5745c780de8a85ce8d2ff12. Thanks!

**Comment by jdevries3133** [comment]
I have made the requested changes; please review again

**Pull request merged**